commit 88763a5cf80ca59a7c3bea32681ce8f697d9995f
Author: Daniel Lezcano <daniel.lezcano@linaro.org>
Date:   Tue Jun 26 12:53:29 2018 +0200

    powercap / idle_inject: Add an idle injection framework
    
    Initially, the cpu_cooling device for ARM was changed by adding a new
    policy inserting idle cycles. The intel_powerclamp driver does a
    similar action.
    
    Instead of implementing idle injections privately in the cpu_cooling
    device, move the idle injection code in a dedicated framework and give
    the opportunity to other frameworks to make use of it.
    
    The framework relies on the smpboot kthreads which handles via its
    main loop the common code for hotplugging and [un]parking.
    
    This code was previously tested with the cpu cooling device and went
    through several iterations. It results now in split code and API
    exported in the header file. It was tested with the cpu cooling device
    with success.
    
    Signed-off-by: Daniel Lezcano <daniel.lezcano@linaro.org>
    Reviewed-by: Viresh Kumar <viresh.kumar@linaro.org>
    [ rjw: Rewrite of all comments ]
    Signed-off-by: Rafael J. Wysocki <rafael.j.wysocki@intel.com>

diff --git a/drivers/powercap/Makefile b/drivers/powercap/Makefile
index 0a21ef31372b..1b328854b36e 100644
--- a/drivers/powercap/Makefile
+++ b/drivers/powercap/Makefile
@@ -1,2 +1,3 @@
 obj-$(CONFIG_POWERCAP)	+= powercap_sys.o
 obj-$(CONFIG_INTEL_RAPL) += intel_rapl.o
+obj-$(CONFIG_IDLE_INJECT) += idle_inject.o